weave/trace_server/trace_server_interface.py

```diff
@@ -318,6 +319,75 @@ class FileContentReadRes(BaseModel):
     content: bytes
 
 
+class FeedbackPayloadReactionReq(BaseModel):
+    emoji: str
+
+
+class FeedbackPayloadNoteReq(BaseModel):
+    note: str = Field(min_length=1, max_length=1024)
+
+
+class FeedbackCreateReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    weave_ref: str = Field(examples=["weave:///entity/project/object/name:digest"])
+    creator: typing.Optional[str] = Field(default=None, examples=["Jane Smith"])
+    feedback_type: str = Field(examples=["custom"])
+    payload: typing.Dict[str, typing.Any] = Field(
+        examples=[
+            {
+                "key": "value",
+            }
+        ]
+    )
+
+
+class FeedbackCreateReqForInsert(FeedbackCreateReq):
+    wb_user_id: str
+
+
+# The response provides the additional fields needed to convert a request
```

Comment:
what do you mean by this? Is there a back-and-forth here?

Reply:
The create request has a "partial" feedback object, the server does things like assign the id, determine the creation timestamp, possibly augment the payload (e.g. in the emoji case we augment it with the shortcode). So the server response includes all of the fields that are needed to make a complete "Feedback" object that will match what you would get if you had queried for all of the feedback. 

So e.g. in the Python SDK, the response from creation allows us to update the list of feedback without having to do a separate roundtrip query or complete refresh.

Reply:
I see - makes sense